src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -356,74 +367,95 @@ IndexerConnector::IndexerConnector(
+                    builderBulkDelete(bulkData, id, m_indexName);
                 }
-                else
+                m_db->delete_(id);
+            }
+            else
+            {
+                const auto dataString = parsedData.at("data").dump();
+                if (!noIndex)
                 {
-                    const auto dataString = parsedData.at("data").dump();
-                    if (!noIndex)
-                    {
-                        builderBulkIndex(bulkData, id, m_indexName, dataString);
-                    }
-                    m_db->put(id, dataString);
+                    builderBulkIndex(bulkData, id, m_indexName, dataString);
                 }
+                m_db->put(id, dataString);
             }
+        }
 
-            if (!bulkData.empty())
-            {
-                // Process data.
-                HTTPRequest::instance().post(
-                    HttpURL(url),
-                    bulkData,
-                    [](const std::string& response) { logDebug2(IC_NAME, "Response: %s", response.c_str()); },
-                    [](const std::string& error, const long statusCode)
-                    {
-                        logError(IC_NAME, "%s, status code: %ld", error.c_str(), statusCode);
-                        throw std::runtime_error(error);
-                    },
-                    "",
-                    DEFAULT_HEADERS,
-                    secureCommunication);
-            }
-        },
-        DATABASE_BASE_PATH + m_indexName,
-        ELEMENTS_PER_BULK);
+        if (!bulkData.empty())
+        {
+            // Process data.
+            HTTPRequest::instance().post(
+                HttpURL(url),
+                bulkData,
+                [](const std::string& response) { logDebug2(IC_NAME, "Response: %s", response.c_str()); },
+                [](const std::string& error, const long statusCode)
+                {
+                    logError(IC_NAME, "%s, status code: %ld", error.c_str(), statusCode);
+                    throw std::runtime_error(error);
+                },
+                "",
+                DEFAULT_HEADERS,
+                secureCommunication);
+        }
+    };
+
+    try
+    {
+        m_dispatcher =
+            std::make_unique<ThreadDispatchQueue>(functor, DATABASE_BASE_PATH + m_indexName, ELEMENTS_PER_BULK);
+    }
+    catch (std::system_error& e)
+    {
+        // Only try to repair IO and Corruption errors
+        if (e.code() == std::errc::io_error)
+        {
+            Utils::RocksDBWrapper::repairDB(DATABASE_BASE_PATH + m_indexName);
+            m_dispatcher =
+                std::make_unique<ThreadDispatchQueue>(functor, DATABASE_BASE_PATH + m_indexName, ELEMENTS_PER_BULK);
+            logWarn(IC_NAME,
+                    "Database '%s%s' was repaired because it was corrupt.",
+                    DATABASE_BASE_PATH,
+                    m_indexName.c_str());
+        }
+        else
+        {
+            throw;
```

Comment:
We're re-throwing the exception as it captures the context effectively, it could be beneficial to add logging before re-throwing.